### PR TITLE
Use FileSystem and Path for loading data.

### PR DESCRIPTION
**Pull request by JakeWharton**
The compiler is still a convoluted mess that needs fixed.

### CODE REVIEW

File: wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java [hunk 1]
```diff
@@ -68,14 +83,16 @@ public static void main(String... args) throws IOException {
...
-    Schema schema;
-    try {
-      schema = loader.load(options.sourceFileNames);
-    } catch (IOException e) {
-      throw new WireException(e);
+  public void compile() throws IOException {
+    List<String> sourceFileNames = options.sourceFileNames;
+    List<Path> sourcePaths = new ArrayList<>(sourceFileNames.size());
+    for (String sourceFileName : sourceFileNames) {
+      Path path = fs.getPath(sourceFileName);
```
Comment (swankjesse) [inline wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java]: This is relative, right?

File: wire-schema/src/main/java/com/squareup/wire/schema/Loader.java [hunk 2]
```diff
@@ -28,86 +30,58 @@
 
 /** Load proto files and their transitive dependencies, parse them, and link them together. */
 public final class Loader {
-  private final IO io;
-  private final Map<String, ProtoFile> loaded = new LinkedHashMap<>();
+  private final List<Path> dirs;
```
Comment (swankjesse) [inline wire-schema/src/main/java/com/squareup/wire/schema/Loader.java]: paths?

File: wire-schema/src/main/java/com/squareup/wire/schema/Loader.java [hunk 3]
```diff
@@ -28,86 +30,58 @@
...
       return;
     }
 
-    Location location;
-    try {
-      location = io.locate(path);
-    } catch (IOException e) {
-      throw new IOException("Failed to locate " + path + " with " + io, e);
+    ProtoFileElement element = null;
+    for (Path dir : dirs) {
```
Comment (swankjesse) [inline wire-schema/src/main/java/com/squareup/wire/schema/Loader.java]: ahhh, that `paths` is taken!

Comment (swankjesse) [inline wire-schema/src/main/java/com/squareup/wire/schema/Loader.java]: Maybe name 'em roots ?

**Comment by swankjesse** [comment]
LGTM

File: wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java (hunk 1 above)
Comment (JakeWharton) [inline wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java]: Yep

File: wire-schema/src/main/java/com/squareup/wire/schema/Loader.java (hunk 3 above)
Comment (JakeWharton) [inline wire-schema/src/main/java/com/squareup/wire/schema/Loader.java]: We use the term "root" to denote types. I expect this class to undergo significant change in a follow-up I have started.

Comment (JakeWharton) [inline wire-schema/src/main/java/com/squareup/wire/schema/Loader.java]: I renamed to `protoDirs` and `protoFiles` for now. We can bikeshed the names in the follow-up where we'll expose these names in public API.

**Pull request merged**